packages/entities/entities-gateway-services/src/components/GatewayServiceForm.vue

```diff
@@ -830,7 +1129,25 @@ const saveFormData = async (): Promise<AxiosResponse | undefined> => {
 
     return response
   } catch (error: any) {
-    form.errorMessage = getMessageFromError(error)
+    const { fields, messages } = getErrorFieldsFromError(error)
+    form.errorMessages = messages
+
+    if (fields.length) {
+      // display error for each of the fields
+      fields.forEach((errorField) => {
+        const field = errorField.field as string
+        if (field === 'client_certificate.id') {
+          form.formFieldErrors.client_certificate = errorField.message
+        } else if (field === 'ca_certificates[0]') {
```

Comment:
this doesn't seem safe. Why are you assuming existence of `ca_certificates[0]`? What happens if they link multiple CA certs and it's not the first one that has the error?